runtimebp/internal/maxprocs/maxprocs.go

```
@@ -0,0 +1,111 @@
+	// ensure consistent performance.
+	//
+	// This value was NOT chosen scientifically and IS subject to change.
+	defaultCPURequestScale = 1.5
+
+	setByGOMAXPROCS = "gomaxprocs"
+	setByRequest    = "cpu_request"
+)
+
+var (
+	mEnvGOMAXPROCS      = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "baseplate_go_env_gomaxprocs"}, []string{"present"})
+	mEnvCPURequest      = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "baseplate_go_env_baseplate_cpu_request"}, []string{"present"})
+	mEnvCPURequestScale = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "baseplate_go_env_baseplate_cpu_request_scale"}, []string{"present"})
+
+	initialGOMAXPROCS = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "baseplate_go_initial_gomaxprocs",
+	}, []string{"set_by"})
+
+	_ = promauto.NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "baseplate_go_current_gomaxprocs",
+	}, currentGOMAXPROCS)
+)
+
+// Set configures the runtime's GOMAXPROCS using the following heuristic:
+//   1. If $GOMAXPROCS is set, Set relinquishes control to the Go runtime.
+//      This should cause the runtime to respect this value directly.
+//   2. If $BASEPLATE_CPU_REQUEST is unset/invalid, Set relinquishes control to the Go runtime.
+//      This should cause the runtime to use the detected number of CPUs on this machine.
+//   3. Otherwise, $BASEPLATE_CPU_REQUEST is multiplied by $BASEPLATE_CPU_REQUEST_SCALE
+//      (or defaultCPURequestScale) to compute the new GOMAXPROCS, minimum 2.
+//
+// Set also exports several metrics to facilitate further tuning/analysis.
+func Set() {
+	envGOMAXPROCS := &floatEnv{key: "GOMAXPROCS", gauge: mEnvGOMAXPROCS}
+	envCPURequest := &floatEnv{key: "BASEPLATE_CPU_REQUEST", gauge: mEnvCPURequest}
+	envCPURequestScale := &floatEnv{key: "BASEPLATE_CPU_REQUEST_SCALE", gauge: mEnvCPURequestScale}
+
+	for _, env := range []*floatEnv{envGOMAXPROCS, envCPURequest, envCPURequestScale} {
+		env.raw, env.present = os.LookupEnv(env.key)
+		env.val, _ = strconv.ParseFloat(env.raw, 64)
```

is there a good reason to ignore error here?

---

Added a comment